### ISSUE TITLE
move entities in nested-set

### ISSUE THREAD

**Issue by Kampfmoehre** [typeorm/typeorm#4174]
**Issue type:**

[x] question
[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
Hi, we are using typeorm in some of our projects and we are mostly happy with it ;-). However, we found, that in nested-set tree entities there is no moving functionality. One could load an entity, replace the parent with another entity and the update query would correctly update the parentId but would not touch the nsleft and nsright column. As my (admittedly limited) understanding is, these columns should at least be updated when the moved entity has children.

My question is, does this break any tree functionality like the findDescendantsTree method or can it safely be ignored?
Should there be some update to nsleft and nsright similar to the [query that is fired before inserts](https://github.com/typeorm/typeorm/blob/0.2.17/src/persistence/tree/NestedSetSubjectExecutor.ts#L24) in nested sets?

I looked at the code and the tests for tree repositories and entities and saw, that there is an "implement moving todo". As I understand, new features are currently not implemented. Can anybody shed some light on this?

Edit: This is possibly related to https://github.com/typeorm/typeorm/pull/3556 and https://github.com/typeorm/typeorm/issues/2418.

**Comment by fullofcaffeine** [comment]
I'd also like to know the state of this, as I'm about to start using the `TreeRepository`. Any updates?

**Comment by Kampfmoehre** [comment]
We acutally have two cases of tree repositories. One we migrated to materialized-path which has the same problem of missing moving functionality. We come with our own code, see [this issue](https://github.com/typeorm/typeorm/issues/2418#issuecomment-649395585).

As for the other one, maybe we also migrate to another tree repository since we need more than one root and as we discovered this is nothing nested-set can handle.

I cannot give you something other than split your moving into deleting the old entry and creating a new entry. That is what we had done in the meantime.
